src/main/scala/org/constellation/consensus/EdgeProcessor.scala

```diff
@@ -216,88 +216,86 @@ object EdgeProcessor extends StrictLogging {
         .toArray,
-      info.snapshot.snapshot.checkpointBlocks
+      snapshotCheckpointBlocks = info.snapshot.snapshot.checkpointBlocks
         .grouped(chunkSize)
         .map(t => chunkSerialize(t, SnapshotInfoChunk.CHECKPOINT_BLOCKS.name))
         .toArray,
-      info.snapshot.snapshot.publicReputation
+      snapshotPublicReputation = info.snapshot.snapshot.publicReputation
         .grouped(chunkSize)
         .map(t => chunkSerialize(t.toSeq, SnapshotInfoChunk.PUBLIC_REPUTATION.name))
         .toArray,
-      info.acceptedCBSinceSnapshot
+      acceptedCBSinceSnapshotHashes = info.acceptedCBSinceSnapshotHashes
         .grouped(chunkSize)
-        .map(t => chunkSerialize(t, SnapshotInfoChunk.ACCEPTED_CBS_SINCE_SNAPSHOT.name))
+        .map(t => chunkSerialize(t, SnapshotInfoChunk.ACCEPTED_CBS_SINCE_SNAPSHOT_HASHES.name))
         .toArray,
-      info.acceptedCBSinceSnapshotCache
+      acceptedCBSinceSnapshotCache = info.acceptedCBSinceSnapshotCache
         .grouped(chunkSize)
         .map(t => chunkSerialize(t, SnapshotInfoChunk.ACCEPTED_CBS_SINCE_SNAPSHOT_CACHE.name))
         .toArray,
-      info.awaitingCbs
+      awaitingCbs = info.awaitingCbs
         .grouped(chunkSize)
         .map(t => chunkSerialize(t.toSeq, SnapshotInfoChunk.AWAITING_CBS.name))
         .toArray,
-      Array(KryoSerializer.serialize[Int](info.lastSnapshotHeight)),
-      info.snapshotHashes
+      lastSnapshotHeight = Array(KryoSerializer.serialize[Int](info.lastSnapshotHeight)),
+      snapshotHashes = info.snapshotHashes
         .grouped(chunkSize)
         .map(t => chunkSerialize(t, SnapshotInfoChunk.SNAPSHOT_HASHES.name))
         .toArray,
-      info.addressCacheData.toSeq
+      addressCacheData = info.addressCacheData.toSeq
         .grouped(chunkSize)
         .map(partitionMap => chunkSerialize(partitionMap, SnapshotInfoChunk.ADDRESS_CACHE_DATA.name))
         .toArray,
-      info.tips
+      tips = info.tips
         .grouped(chunkSize)
         .map(partitionMap => chunkSerialize(partitionMap.toSeq, SnapshotInfoChunk.TIPS.name))
         .toArray,
-      info.snapshotCache.grouped(chunkSize).map(t => chunkSerialize(t, SnapshotInfoChunk.SNAPSHOT_CACHE.name)).toArray,
-      info.lastAcceptedTransactionRef
+      snapshotCache = info.snapshotCache.grouped(chunkSize).map(t => chunkSerialize(t, SnapshotInfoChunk.SNAPSHOT_CACHE.name)).toArray,
+      lastAcceptedTransactionRef = info.lastAcceptedTransactionRef
         .grouped(chunkSize)
         .map(partitionMap => chunkSerialize(partitionMap.toSeq, SnapshotInfoChunk.LAST_ACCEPTED_TX_REF.name))
         .toArray
     )
+
+  def writeLocal(overWritePath: String) = {
+    val heightPrefix = SnapshotService.snapshotInfoFileHeightPrefix(lastSnapshotHeight)
+    val hash = snapshot.snapshot.hash
+    val path = File(overWritePath.concat(s"/${heightPrefix}$hash/")).createDirectoryIfNotExists()
+    this.toSnapshotInfoSer().writeLocal(basePath = path.pathAsString)
+  }
 }
```

Comment:
Why we create methods which are used only in tests instead of mocking real ones? Imo we should mock it and in general not focus on E2E right now as we want to test the redownload manually anyway :P